language-support/scala/codegen/src/main/scala/com/digitalasset/lf/codegen/CodeGen.scala

```diff
@@ -137,49 +124,34 @@ object CodeGen {
   private def combineInterfaces(dar: Dar[Interface]): EnvironmentInterface =
     EnvironmentInterface.fromReaderInterfaces(dar)
 
-  private def combineEnvInterfaces(as: NonEmptyList[EnvironmentInterface]): EnvironmentInterface =
-    as.suml1
+  private def templateCount(interface: EnvironmentInterface): Int =
+    interface.typeDecls.count {
+      case (_, InterfaceType.Template(_, _)) => true
+      case _ => false
+    }
 
   private def packageInterfaceToScalaCode(util: LFUtil): Unit = {
-    val interface = util.iface
-
-    val orderedDependencies
-        : OrderedDependencies[Identifier, TypeDeclOrTemplateWrapper[DefTemplateWithRecord]] =
-      DependencyGraph.orderedDependencies(util.iface)
-    val (templateIds, typeDeclsToGenerate): (
-        Map[Identifier, DefTemplateWithRecord],
-        List[ScopedDataType.FWT],
-    ) = {
-
-      /* Here we collect templates and the
-       * [[TypeDecl]]s without generating code for them.
-       */
-      val templateIdOrTypeDecls
-          : List[(Identifier, DefTemplateWithRecord) Either ScopedDataType.FWT] =
-        orderedDependencies.deps.toList.flatMap {
-          case (templateId, Node(TypeDeclWrapper(typeDecl), _, _)) =>
-            Seq(Right(ScopedDataType.fromDefDataType(templateId, typeDecl)))
-          case (templateId, Node(TemplateWrapper(templateInterface), _, _)) =>
-            Seq(Left((templateId, templateInterface)))
-        }
-
-      templateIdOrTypeDecls.partitionMap(identity).leftMap(_.toMap)
```

Comment:
Re: https://github.com/digital-asset/daml/issues/13764

This code has been moved to the `transitiveClosure` method (to be shared with the Java codegen) and changed slightly. It seems innocuous as well but another change worth looking into, I believe.